app/frontend/actions.ts

```diff
@@ -479,28 +493,33 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
 
   /* SEND ADA */
 
-  const setTransactionSummary = (tab, plan, coins?, donationAmount?) => {
+  const setTransactionSummary = (
+    tab, // TODO: add enum for tabs
+    plan: TxPlan,
+    coins?: Lovelace,
+    donationAmount?: Lovelace
+  ) => {
     setState({
       sendTransactionSummary: {
-        amount: coins || 0,
-        donation: donationAmount || 0,
-        fee: plan.fee != null ? plan.fee : plan.estimatedFee,
-        plan: plan.fee != null ? plan : null,
+        amount: coins || (0 as Lovelace),
+        donation: donationAmount || (0 as Lovelace),
+        fee: plan.fee,
+        plan,
         tab,
```

Comment:
wtf is tab doing in transaction summary?

Reply:
I didnt refactor this, but its rather a bad naming, it should be `type` and `TxType` enum.